torch/testing/_internal/opinfo/core.py

```diff
@@ -1305,21 +1305,18 @@ def get_decorators(self, test_class, test_name, device, dtype, param_kwargs):
         return result
 
     def supported_dtypes(self, device_type):
-        if device_type == "cpu":
-            return self.dtypes
+        device_type = str(device_type).split(":", 1)[0]
```

Comment:
Should we rather do

```suggestion
        device_type = torch.device(device_type).type
```

? This way, we get the default handling of unknowns by the `torch.device` constructor. As written "foo:bar" would be acceptable. `torch.device` correctly fails.